databroker/mongo_normalized.py

```diff
@@ -2241,6 +2242,11 @@ def default_validate_shape(key, data, expected_shape):
         else:  # margin == 0
             padding.append((0, 0))
     padded = numpy.pad(data, padding, "edge")
+
+    logger.warning(f"The data.shape: {data.shape} did not match the expected_shape: "
```

Comment:
Is there a way that I can add the run uid to this warning?
`default_validate_shape` doesn't have `self` so something like this wont work.
```
f"RunStart UID: {self._run.metadata()['start']['uid']!r}"
```

Reply:
The signature of `validate_metadata_shape` could be modified to accept the `uid` (last, and optionally, for back-compat).